### PR TITLE
Eliminate busy wait, fixing 100% CPU usage.

### PR DESCRIPTION
**Pull request by tomquisel**
Also remove unecessary communicate() call. It just prints "None" to the
log.

### CODE REVIEW

**Comment by Tarrasch** [comment]
Erm, I don't see why this PR was frogotten @tomquisel . But you'll have to squash the commits I think. Then I can merge once somebody else +1's. I don't know much about spark.

By the way, the commit header must name "spark" or something somewhere, as the commit message is unclear. :)

File: test/contrib/spark_test.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def test_app_interruption(self, proc):
         def interrupt():
             raise KeyboardInterrupt()
 
-        proc.return_value.poll = interrupt
+        proc.return_value.wait = interrupt
```
Comment (erikbern) [inline test/contrib/spark_test.py]: btw ideally we should use the mock library for this. but that's prob outside the scope of this PR

**Comment by Tarrasch** [comment]
I wish you didn't merge this as I think the commit-quality was pretty poor :-/

**Comment by erikbern** [comment]
I think it was fine. What was the issue? Not squashed? yeah maybe should have required squashing it. But I also think it would have been a bit rude to get back after almost a month of silence then asking for a rebase

**Comment by tomquisel** [comment]
Totally not a big deal either way :). I'll squash commits and make the commit messages more clear in the future.

**Comment by Tarrasch** [comment]
Sure, sorry for being not so responsive on the PR, don't hesitate to ping us @tomquisel :)

@erikbern My main issue is that the commit message doesn't mention "spark" *anywhere*. It sounds like it's fixing a critical bug in the scheduler or something. :)

**Comment by erikbern** [comment]
yeah that's a good point. i was actually really confused too first :)

**Pull request merged**